File: app/src/main/kotlin/tech/mmarca/openvitals/core/preferences/ActivityWeekMode.kt
package tech.mmarca.openvitals.core.preferences

import tech.mmarca.openvitals.core.period.WeekPeriodMode

enum class ActivityWeekMode {
    MONDAY_TO_SUNDAY,
    LAST_7_DAYS,
}

fun ActivityWeekMode.toWeekPeriodMode(): WeekPeriodMode =
    when (this) {
        ActivityWeekMode.MONDAY_TO_SUNDAY -> WeekPeriodMode.MONDAY_TO_SUNDAY
        ActivityWeekMode.LAST_7_DAYS -> WeekPeriodMode.LAST_7_DAYS
    }

File: lib/domain/preferences/activity_week_mode.dart
import '../../core/period/time_range.dart';

enum ActivityWeekMode {
  mondayToSunday,
  last7Days,
}

extension ActivityWeekModeMapping on ActivityWeekMode {
  WeekPeriodMode toWeekPeriodMode() {
    switch (this) {
      case ActivityWeekMode.mondayToSunday:
        return WeekPeriodMode.mondayToSunday;
      case ActivityWeekMode.last7Days:
        return WeekPeriodMode.last7Days;
    }
  }
}
